### PR TITLE
fix: update guild iterator + scheduled event user iterator

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Noticed two bugs while refactoring parts of the code. They are not an issue anymore in the upcoming refactor, but having a separate fix makes it easier to backport to older versions.

- guild iterator
    - Only use `before` strategy if `before` parameter was specified
    - Don't stop after first page if `limit=None` is set
    - Always return results in correct order
- scheduled event user iterator
    - Fix missing import

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/iterators.py [hunk 1]
```diff
@@ -688,9 +689,11 @@ def create_guild(self, data):
     async def fill_guilds(self):
         if self._get_retrieve():
             data = await self._retrieve_guilds(self.retrieve)
-            if self.limit is None or len(data) < 100:
+            if len(data) < 100:
```
Comment (onerandomusername) [inline disnake/iterators.py:692]: shouldn't this be 200, as the endpoint now provides up to 200 guilds?

https://discord.com/developers/docs/resources/user#get-current-user-guilds

**Review by onerandomusername — COMMENTED** [review]
Could you also add to the issue body why these changes are being made, and what they fix?

File: disnake/iterators.py (hunk 1 above)
Comment (shiftinv) [reply disnake/iterators.py]: aefe260be307465e0a90116ec422417d5dcfdbed

**Review by onerandomusername — APPROVED** [review]
lgtm

**Pull request merged**